### PR TITLE
Grafana/InfluxDB installation enhancements

### PR DESCRIPTION
**Pull request by holgerfriedrich**
* Timeouts for service start #760
* Grafana admin password length is enforced #562
* Local installation of InfluxDB now also patches existing installations
* Password reset for existing Grafana and local InfluxDB installations

Signed-off-by: Holger Friedrich <mail@holger-friedrich.de>

### CODE REVIEW

**Comment by mstormi** [comment]
Can you add checking for memory <=512MB (or check for is_pizero || is_pizerow || is_pione || is_cmone) and refuse to install InfluxDB+Grafana in that case to avoid users to overload their box ?
You could maybe output something like "if you still insist to do it, you can change the openHABian source code to do it for" so we're reasonably safe novices won't try but pros aren't p**d off.

**Comment by mstormi** [comment]
Also change the error message in case installs fail to match that from the end of menu.bash

**Comment by holgerfriedrich** [comment]
@mstormi done

File: functions/influxdb+grafana.bash [hunk 1]
```diff
@@ -29,15 +29,26 @@ influxdb_grafana_setup() {
   local password_check
   local text_grafana_admin_password
   local text_openHAB_integration
+  local totalmemory
+  local lowmemory
 
   FAILED=0
-  text_intro="This will install and configure InfluxDB and Grafana. For more information please consult this discussion thread:\\nhttps://community.openhab.org/t/13761/1"
+  text_intro="This will install and configure InfluxDB and Grafana. For more information please consult this discussion thread:\\nhttps://community.openhab.org/t/13761/1\\n\\nNOTE for existing installations:\\n - Grafana password will be reset and configuration adpated\\n - If local installation of InfluxDB is choosen, passwords will be reset and config files will be changed"
```
Comment (mstormi) [inline functions/influxdb+grafana.bash:36]: typo: "adpated"

File: functions/influxdb+grafana.bash [hunk 2]
```diff
@@ -29,15 +29,26 @@ influxdb_grafana_setup() {
...
+  text_lowmem="WARNING: InfluxDB and Grafana tend to use a lot of memory. Your machine reports less than 1 GB memory, and we STRICTLY RECOMMEND NOT TO PROCEED!\\n\\nDISCLAIMER: Proceed on your own risk, do NOT report tickets if you run in problems."
```
Comment (mstormi) [inline functions/influxdb+grafana.bash:37]: Proceed _at_ your own risk _and_ do NOT ....

File: functions/influxdb+grafana.bash [hunk 3]
```diff
@@ -29,15 +29,26 @@ influxdb_grafana_setup() {
...
   text_fail="Sadly there was a problem setting up the selected option. Please report this problem in the openHAB community forum or as a openHABian GitHub issue."
+  text_fail_lowmem="Sadly there was a problem setting up the selected option. Your machine reports less than 1 GB memory, please consider upgrading your hardware for Garafana/InfluxDB."
```
Comment (mstormi) [inline functions/influxdb+grafana.bash:39]: "Garafana"

File: functions/influxdb+grafana.bash (hunk 2 above)
Comment (holgerfriedrich) [reply functions/influxdb+grafana.bash:37]: ok

**Comment by mstormi** [comment]
I believe you forgot the "Fixes #nnn" so you have to manually close those issues now

**Comment by holgerfriedrich** [comment]
@mstormi I did this intentionally because I wanted to comment to the authors of the issues. Commented, linked to PR, closed.
Thanks for the reminder.

**Pull request merged**